spm_BIDS.m

```diff
@@ -504,7 +504,7 @@
 
         %-Diffusion imaging file
         %------------------------------------------------------------------
-        p = parse_filename(f{i}, {'sub','ses','acq','run', 'bval','bvec'});
+        p = parse_filename(f{i}, {'sub','ses','task','acq','rec','dir','run','part','chunk','bval','bvec'});
```

Comment:
no task in diffusion according to the yml schema:

https://github.com/bids-standard/bids-specification/blob/c7cabedc0870acc454fdc01af223379f797a011f/src/schema/rules/files/raw/dwi.yaml#L14